Votca fails to build with Boost 1.83

The Build fails with:
```console
  In file included from /tmp/votca-20230915-65962-1sgj8rr/votca-2022.1/xtp/src/libxtp/calculators/kmclifetime.cc:24:
  /usr/local/include/boost/progress.hpp:23:3: error: This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
  # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
    ^
  In file included from /tmp/votca-20230915-65962-1sgj8rr/votca-2022.1/xtp/src/libxtp/calculators/kmclifetime.cc:24:
  In file included from /usr/local/include/boost/progress.hpp:29:
  /usr/local/include/boost/timer.hpp:21:3: error: This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
  # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
    ^
  In file included from /tmp/votca-20230915-65962-1sgj8rr/votca-2022.1/xtp/src/libxtp/calculators/neighborlist.cc:20:
  /usr/local/include/boost/progress.hpp:23:3: error: This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
  # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
    ^
  In file included from /tmp/votca-20230915-65962-1sgj8rr/votca-2022.1/xtp/src/libxtp/calculators/neighborlist.cc:20:
  In file included from /usr/local/include/boost/progress.hpp:29:
  /usr/local/include/boost/timer.hpp:21:3: error: This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
  # error This header is deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress this error.)
    ^
  2 errors generated.
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -12,6 +12,7 @@ Version 2023-rc.2-dev
 -  update MKL detection to MKL config (#1053)
 -  remove geometry optimization (#1077)
 -  remove all references to old pdf manual (#1081)
+-  replaced deprecated boost::progress headers (#1083)
 
 Version 2023-rc.2 (released 02.06.23)
 =====================================

--- a/xtp/src/libxtp/calculators/kmclifetime.cc
+++ b/xtp/src/libxtp/calculators/kmclifetime.cc
@@ -21,7 +21,7 @@
 
 // Third party includes
 #include <boost/format.hpp>
-#include <boost/progress.hpp>
+#include <boost/timer/progress_display.hpp>
 
 // VOTCA includes
 #include <votca/tools/constants.h>
@@ -290,7 +290,7 @@ void KMCLifetime::RunVSSM() {
 
       // determine where it will jump to
       ResetForbiddenlist(forbiddendests);
-      boost::progress_display progress(insertions_);
+      boost::timer::progress_display progress(insertions_);
 
       while (true) {
         // LEVEL 2

--- a/xtp/src/libxtp/calculators/neighborlist.cc
+++ b/xtp/src/libxtp/calculators/neighborlist.cc
@@ -17,7 +17,7 @@
 
 // Third party includes
 #include <boost/format.hpp>
-#include <boost/progress.hpp>
+#include <boost/timer/progress_display.hpp>
 
 // Local private VOTCA includes
 #include "neighborlist.h"
@@ -124,7 +124,7 @@ bool Neighborlist::Evaluate(Topology& top) {
 
   top.NBList().Cleanup();
 
-  boost::progress_display progress(segs.size());
+  boost::timer::progress_display progress(segs.size());
   // cache approx sizes
   std::vector<double> approxsize = std::vector<double>(segs.size(), 0.0);
 #pragma omp parallel for

